core/database/foxx/api/repo_router.js

```diff
@@ -722,13 +759,34 @@ router
                         subject_id = req.queryParams.subject;
                     else subject_id = g_lib.getUserFromClientID(req.queryParams.subject)._id;
 
-                    var result = g_tasks.taskInitAllocDelete(
-                        client,
-                        req.queryParams.repo,
-                        subject_id,
-                    );
+                    // Find the repository using the new type system
+                    var findResult = RepositoryOps.find(req.queryParams.repo);
+                    if (!findResult.ok) {
+                        throw [findResult.error.code, findResult.error.message];
+                    }
+
+                    var repository = findResult.value;
+
+                    // Check permissions
+                    var permResult = RepositoryOps.checkPermission(repository, client._id, "admin");
+                    if (!permResult.ok || !permResult.value) {
+                        throw g_lib.ERR_PERM_DENIED;
+                    }
+
+                    // Check if subject exists
+                    if (!g_db._exists(subject_id)) {
+                        throw [g_lib.ERR_NOT_FOUND, "Subject not found: " + subject_id];
```

Comment:
It would be clearer if we include in the exeception, that the allocation delete request failed, for audit reasons, in cases around authorization it is really useful to know who attempted to do what.